src/gflownet/envs/graph_building_env.py

```diff
@@ -325,6 +330,32 @@ def reverse(self, g: Graph, ga: GraphAction):
             return GraphAction(GraphActionType.RemoveNodeAttr, source=ga.source, attr=ga.attr)
         if ga.action == GraphActionType.SetEdgeAttr:
             return GraphAction(GraphActionType.RemoveEdgeAttr, source=ga.source, target=ga.target, attr=ga.attr)
+        if ga.action == GraphActionType.RemoveNode:
+            # TODO: implement neighbors or something?
+            # neighbors = list(g.neighbors(ga.source))
+            # source = 0 if not len(neighbors) else neighbors[0]
+            neighbors = [i for i in g.edges if i[0] == ga.source or i[1] == ga.source]
+            assert len(neighbors) <= 1  # RemoveNode should only be a legal action if the node has one or zero neighbors
+            source = 0 if not len(neighbors) else neighbors[0][0] if neighbors[0][0] != ga.source else neighbors[0][1]
+            return GraphAction(GraphActionType.AddNode, source=source, value=g.nodes[ga.source]["v"])
+        if ga.action == GraphActionType.RemoveEdge:
+            return GraphAction(GraphActionType.AddEdge, source=ga.source, target=ga.target)
+        if ga.action == GraphActionType.RemoveNodeAttr:
+            return GraphAction(
+                GraphActionType.SetNodeAttr,
+                source=ga.source,
+                target=ga.target,
+                attr=ga.attr,
+                value=g.nodes[ga.source][ga.attr],
+            )
+        if ga.action == GraphActionType.RemoveEdgeAttr:
```

Comment:
Why a serie of `if` instead of `if` + `elif` (and also an `else` as a sanity check)?

By doing so, you'll likely decrease the compute time spent doing all those checks.

Reply:
Ah yes good catch